src/java.base/share/classes/java/io/DataInputStream.java

```diff
@@ -33,9 +33,10 @@
  * way. An application uses a data output stream to write data that
  * can later be read by a data input stream.
  * <p>
- * DataInputStream is not necessarily safe for multithreaded access.
- * Thread safety is optional and is the responsibility of users of
- * methods in this class.
+ * A DataInputStream is not safe for use by multiple concurrent
+ * threads. If a DataOutputStream is to be used by more than one
+ * thread then access to the data output stream should be controlled
+ * by appropriate synchronization.
```

Comment:
Thanks for the update. Looks good except the s/DataOutputStream/DataInputStream/ and s/output/input/ here.